src/org/zaproxy/zap/extension/pscanrules/TestInfoSessionIdURL.java

```diff
@@ -147,88 +152,56 @@ public void scanHttpRequestSend(HttpMessage msg, int id) {
-        // we search for all params looking for one session element
-        // maybe more efficient because we don't build Patterns any time
-        //String[] params = msg.getParamNames();
-
-        // The Session ID list option param
+        TreeSet<HtmlParameter> urlParams = msg.getUrlParams();
+        
+        if (urlParams.isEmpty()) {
+    		return; //No params, no need to proceed
+    	}
+        
+        String uri = msg.getRequestHeader().getURI().toString();
+        
+        // The Session ID list from option param (panel)
         OptionsParam options = Model.getSingleton().getOptionsParam();
-        HttpSessionsParam sessionOptions =
-                options.getParamSet(HttpSessionsParam.class);
-
-        // Loop on all possible 
-        // session id variables (looking all along the url)
-        // -----------------------------------------------------
-        // We've to rebuild every time the patterns because
-        // the user could change options during the session
-        // so we have to be sure that we search for the 
-        // session ids that have really been selected (or added)
-        // -----------------------------------------------------
-        for (String sessionid : sessionOptions.getDefaultTokensEnabled()) {
-            pattern = Pattern.compile("(\\Q" + sessionid + "\\E)=[^\\&]+", Pattern.CASE_INSENSITIVE);
-            matcher = pattern.matcher(uri);
-
-            if (matcher.find()) {
-                // Get the overall sessionvar=value pattern
-                sessionIdPair = matcher.group(0);
-                // Get the value portion
-                sessionIdValue = matcher.group(0).split("=")[1];
-                // Get the sessionvar name
-                sessionIdName = matcher.group(1);
-
-                // In passive mode there not exists any KB available
-                // This was the old implementation
-                // --------------------------------------------------
-                //String kb = getKb().getString("sessionId/nameValue");
-                //if (kb == null || !kb.equals(sessionIdValue)) {
-                //    getKb().add("sessionId/nameValue", sessionIdValue);
-                //    bingo(Alert.RISK_LOW, Alert.WARNING, uri, null, "", null, sessionIdValue, base);
-                //}                
-                //kb = getKb().getString("sessionId/name");
-                //getKb().add("sessionId/name", sessionIdName);
-
-                if (sessionIdValue.length() > SESSION_TOKEN_MIN_LENGTH) { 
-	                
+        HttpSessionsParam sessionOptions = options.getParamSet(HttpSessionsParam.class);
+        List<String> sessionIds = sessionOptions.getDefaultTokensEnabled();
+        for (HtmlParameter param: urlParams) { //Iterate through the parameters
+        	//If the parameter name is one of those on the Session Token list from the options panel
+        	if (sessionIds.contains(param.getName().toLowerCase())) { 

```

Comment:
Although it shouldn't make much difference in this case, consider converting to lower case in a locale independent way (i.e. `toLowerCase(Locale.ROOT)`).